fairlearn/metrics/_exposure.py

```diff
@@ -0,0 +1,304 @@
+# Copyright (c) Fairlearn contributors.
+# Licensed under the MIT License.
+
+import numpy as np
+
+from fairlearn.metrics._input_manipulations import _convert_to_ndarray_and_squeeze
+from ._metric_frame import MetricFrame
+
+_EMPTY_INPUT_PREDICTIONS_ERROR_MESSAGE = "Empty {0} passed to exposure function."
+_INVALID_RANKING_ERROR_MESSAGE = "Please input a valid complete ranking."
+_ZERO_DIVISION_ERROR = "Average utility is 0, which causes a zero division error."
+
+
+def exposure(y_true,
+             y_pred,
+             *,
+             sample_weight=None) -> float:
+    """Calculate the exposure allocated to the ranking in y_pred.
+
+    Exposure is defined as the average logarithmic discount, where
+    logarithmic discount equals :math:`1 / log_2 (1 + i)` for i in y_pred, as used
+    in discounted cumulative gain (DCG).
+
+    For consistency with other metric functions, the ``y_true`` argument
+    is required, but ignored.
+
+
+    Parameters
+    ----------
+    y_true : array_like
+        Ground truth relevance scores. (ignored)
+    y_pred : array_like
+        The predicted ranking
+    sample_weight : array_like
+        Optional array of sample weights
+    """
+    if len(y_pred) == 0:
+        raise ValueError(_EMPTY_INPUT_PREDICTIONS_ERROR_MESSAGE.format('y_pred'))
+
+    if len(set(y_pred)) != len(y_pred):  # check for repetition in input ranking
+        raise ValueError(_INVALID_RANKING_ERROR_MESSAGE)
+
+    # ranking should start by 1, to prevent zero division.
+    if 0 in y_pred:
+        y_pred = [x+1 for x in y_pred]
+
+    y_pred = _convert_to_ndarray_and_squeeze(y_pred)
+    v = [1 / np.log2(1 + j) for j in y_pred]  # logarithmic discount
+
+    s_w = np.ones(len(y_pred))
+    if sample_weight is not None:
+        s_w = np.squeeze(np.asarray(sample_weight))
+
+    return np.dot(v, s_w).sum() / len(y_pred)
+
+
+def utility(y_true,
+            y_pred,
+            *,
+            sample_weight=None) -> float:
+    """Calculate the utility of the ranking.
+
+    Utility is defined as the average of y_true.
+
+    The goal of this metric is to be used in the `exposure_utility_ratio` metric. Where we try
+    to keep the `exposure` proportional to the `utility`.
+
+    For consistency with other metric functions, the ``y_pred`` argument
+    is required, but ignored.
+
+
+    Parameters
+    ----------
+    y_true : array_like
+        Ground truth relevance scores.
+    y_pred : array_like
+        The predicted ranking (ignored)
```

Comment:
I think for consistency it is better to mention it since it is a required input parameter. And because it is a ranking metric, I believe it is best to describe y_pred as a ranking.